lib/helpers/PackageUiHelper.ts

```diff
@@ -0,0 +1,157 @@
+  }
+
+  async isTextNoPackagesHaveBeenInstalledVisible() {
+    return await expect(this.noPackagesHaveBeenInstalledTxt).toBeVisible();
+  }
+
+  async clickCreatedTab() {
+    await this.createdTabBtn.click({force: true});
+  }
+
+  async clickInstalledTab() {
+    await this.installedTabBtn.click({force: true});
+  }
+
+  async clickPackagesTab() {
+    await this.packagesTabBtn.click({force: true});
+  }
+
+  async clickChooseBtn() {
+    await this.chooseBtn.click();
+  }
+
+  async isMarketPlaceIFrameVisible() {
+    return await expect(this.marketPlaceIFrame).toBeVisible();
+  }
+
+  async clickCreatePackageButton() {
+    await this.createPackageBtn.click();
+  }
+
+  async enterPackageName(packageName: string) {
+    await this.packageNameTxt.clear();
+    await this.packageNameTxt.fill(packageName);
+  }
+
+  async doesPackageNameExist(packageName: string) {
+    return this.page.getByRole('button', {name: packageName});
+  }
+
+  async clickExistingPackageName(packageName: string) {
+    await this.page.getByRole('button', {name: packageName}).click();
+  }
+
+  async clickDeleteButtonForPackageName(packageName: string) {
+    await this.page.locator('uui-ref-node-package', {hasText: packageName}).getByLabel('Delete package').click({force: true});
+  }
+
+  async clickSaveChangesToPackageButton() {
+    await this.saveChangesToPackageBtn.click();
+  }
+
+  async clickAddContentToPackageButton() {
+    await this.addContentToPackageBtn.click({force: true});
+  }
+
+  async clickAddMediaToPackageButton() {
+    await this.addMediaToPackageBtn.click({force: true});
+  }
+
+  async clickAddDocumentTypeToPackageButton() {
+    await this.addDocumentTypeToPackageBtn.click({force: true});
+  }
+
+  async clickAddMediaTypeToPackageButton() {
+    await this.addMediaTypeToPackageBtn.click({force: true});
+  }
+
+  async clickAddLanguageToPackageButton() {
+    await this.addLanguageToPackageBtn.click({force: true});
+  }
+
+  async clickAddDictionaryToPackageButton() {
+    await this.addDictionaryToPackageBtn.click({force: true});
+  }
+
+  async clickAddDataTypesToPackageButton() {
+    await this.addDataTypesToPackageBtn.click({force: true});
+  }
+
+  async clickAddTemplatesToPackageButton() {
+    await this.addTemplatesToPackagesBtn.click({force: true});
+  }
+
+  async clickAddPartialViewToPackageButton() {
+    await this.addPartialViewToPackageBtn.click({force: true});
+  }
+
+  async clickAddScriptToPackageButton() {
+    await this.addScriptToPackageBtn.click({force: true});
+  }
+
+  async clickAddStylesheetToPackageButton() {
+    await this.addStylesheetToPackageBtn.click({force: true});
+  }
+
+  // Downloads the package and converts it to a string
+  async downloadPackage(packageId: string) {
+    const responsePromise = this.page.waitForResponse(umbracoConfig.environment.baseUrl + '/umbraco/management/api/v1/package/created/' + packageId + '/download');
```

Comment:
It would be beneficial to separate the code that interacts with the API from the code that interacts with the UI. It should be declared in PackageApiHelper

Reply:
This method does not interact with the API. It looks at the network responses on the browser and waits until a response is the same as the one it's waiting for.

Reply:
it makes sense to me now 😄